yt/frontends/exodus_ii/data_structures.py

```diff
@@ -336,7 +336,30 @@ def _read_connectivity(self):
         connectivity = []
         with self._handle.open_ds() as ds:
             for i in range(self.parameters["num_meshes"]):
-                connectivity.append(ds.variables["connect%d" % (i + 1)][:].astype("i8"))
+                var = ds.variables["connect%d" % (i + 1)][:].astype("i8")
+                try:
+                    elem_type = var.elem_type.lower()
+                    if elem_type == "nsided":
+                        arbitrary_polyhedron = True
+                    elif elem_type == "nfaced":
+                        raise NotImplementedError(
+                            "3D arbitrary polyhedra are not implemented yet"
+                        )
+                    else:
+                        arbitrary_polyhedron = False
+                except AttributeError:
+                    arbitrary_polyhedron = False
+
+                conn = var[:].astype("i8")
+                if arbitrary_polyhedron:
+                    nodes_per_element = ds.variables["ebepecnt%d" % (i + 1)]
+                    if np.any(nodes_per_element != nodes_per_element[0]):
+                        raise NotImplementedError("only equal-size polyhedra supported")
+                    conn.shape = (
+                        conn.shape[0] // nodes_per_element[0],
+                        nodes_per_element[0],
+                    )
```

Comment:
Reusing the first element of `nodes_per_element` more than twice grants a variable name imo.
Additionally, and this is probably very nitpicky, but I am not 100% confident in the way the `conn` array is being reshaped, so I'm suggesting refactoring it with a check.
```suggestion
                    npe = nodes_per_element[0]
                    if np.any(nodes_per_element != npe):
                        raise NotImplementedError("only equal-size polyhedra supported")
                    q, r = np.divmod(len(conn), npe)
                    assert r == 0
                    conn.shape = (q, npe)
```

Reply:
I'm fine with this. Is it yt style to prefer `len(conn)` to `conn.shape[0]`. I find the latter a bit clearer.

Reply:
So this suggestion reflects my now understanding here. I am under the impression that this array can only be 1D before we reshape it, in which case I think using `len()` is ok.
Imo using .shape[0] gives the false impression that the array is multidimensional, in which case the reshaping line would be broken (I think).

Reply:
OK, agreed.